barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -2357,45 +2357,71 @@ void AvmTraceBuilder::op_l1_to_l2_msg_exists(uint8_t indirect,
     debug("l1_to_l2_msg_exists side-effect cnt: ", side_effect_counter);
 }
 
-void AvmTraceBuilder::op_get_contract_instance(uint8_t indirect, uint32_t address_offset, uint32_t dst_offset)
+void AvmTraceBuilder::op_get_contract_instance(
+    uint8_t indirect, uint8_t member_enum, uint16_t address_offset, uint16_t dst_offset, uint16_t exists_offset)
 {
+    ASSERT(member_enum < static_cast<int>(ContractInstanceMember::MAX_MEMBER));
+    ContractInstanceMember chosen_member = static_cast<ContractInstanceMember>(member_enum);
+
     auto clk = static_cast<uint32_t>(main_trace.size()) + 1;
 
-    auto [resolved_address_offset, resolved_dst_offset] =
-        Addressing<2>::fromWire(indirect, call_ptr).resolve({ address_offset, dst_offset }, mem_trace_builder);
+    auto [resolved_address_offset, resolved_dst_offset, resolved_exists_offset] =
+        Addressing<3>::fromWire(indirect, call_ptr)
+            .resolve({ address_offset, dst_offset, exists_offset }, mem_trace_builder);
 
     auto read_address = constrained_read_from_memory(
         call_ptr, clk, resolved_address_offset, AvmMemoryTag::FF, AvmMemoryTag::FF, IntermRegister::IA);
     bool tag_match = read_address.tag_match;
 
+    // Read the contract instance
+    ContractInstanceHint instance = execution_hints.contract_instance_hints.at(read_address.val);
+
+    const FF member_value = chosen_member == ContractInstanceMember::DEPLOYER ? instance.deployer_addr
```

Comment:
Well, a switch over the enum would have been nice...but then you "lose" the const I guess.

Reply:
Ah I love the ternary operator :rofl: probably comes from spending too much time in React.